connectors/github-pr/src/test/java/com/vmware/connectors/github/pr/GithubPrControllerTests.java

```diff
@@ -0,0 +1,710 @@
+    }
+
+    private ResultActions requestChanges(String authToken, String request) throws Exception {
+        return perform(
+                setupPostRequest(
+                        "/api/v1/vmware/test-repo/99/request-changes",
+                        APPLICATION_FORM_URLENCODED,
+                        authToken,
+                        request == null ? "" : String.format("request=%s", request)
+                )
+        );
+    }
+
+    /////////////////////////////
+    // Request Cards
+    /////////////////////////////
+
+    @Test
+    public void testRequestCardsUnauthorized() throws Exception {
+        mockGithub.expect(ExpectedCount.manyTimes(), requestTo(any(String.class)))
+                .andRespond(withUnauthorizedRequest());
+
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/valid/cards/card.json"))
+                .andExpect(status().isBadRequest())
+                .andExpect(header().string("X-Backend-Status", "401"));
+    }
+
+    @Test
+    public void testRequestCardsAuthHeaderMissing() throws Exception {
+        requestCards(null, fromFile("requests/valid/cards/card.json"))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testRequestCardsSuccess() throws Exception {
+        trainGithubForCards();
+
+        requestCards(GITHUB_AUTH_TOKEN, fromFile("requests/valid/cards/card.json"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(
+                        content().string(
+                                JsonReplacementsBuilder
+                                        .from(fromFile("responses/success/cards/card.json"))
+                                        .buildForCards()
+                        )
+                );
+    }
+
+    private void trainGithubForCards() throws Exception {
+        mockGithub.expect(requestTo("https://api.github.com/repos/vmware/test-repo/pulls/1"))
+                .andExpect(header(AUTHORIZATION, "Bearer " + GITHUB_AUTH_TOKEN))
+                .andExpect(method(GET))
+                .andRespond(withSuccess(fromFile("fake/cards/small-merged-pr.json"), APPLICATION_JSON));
+
+        mockGithub.expect(requestTo("https://api.github.com/repos/vmware/test-repo/pulls/2"))
+                .andExpect(header(AUTHORIZATION, "Bearer " + GITHUB_AUTH_TOKEN))
+                .andExpect(method(GET))
+                .andRespond(withSuccess(fromFile("fake/cards/small-open-pr.json"), APPLICATION_JSON));
+
+        mockGithub.expect(requestTo("https://api.github.com/repos/vmware/test-repo/pulls/3"))
+                .andExpect(header(AUTHORIZATION, "Bearer " + GITHUB_AUTH_TOKEN))
+                .andExpect(method(GET))
+                .andRespond(withSuccess(fromFile("fake/cards/big-closed-pr.json"), APPLICATION_JSON));
+
+        mockGithub.expect(requestTo("https://api.github.com/repos/vmware/test-repo/pulls/0-not-found"))
```

Comment:
I made this 0-not-found so it sorted ahead of the others.  I originally had a mistake in where I put my error handling (the skip404 code) which was causing it to skip the rest of the requests to get skipped, however, I "lucked out" and my tests didn't fail.  It wasn't until later that I noticed the problem when testing in the iOS simulator, so I fixed the tests, then the bug.